torchvision/datasets/kinetics.py

```diff
@@ -23,4 +24,8 @@ def __getitem__(self, idx):
         video, audio, info, video_idx = self.video_clips.get_clip(idx)
         label = self.samples[video_idx][1]
 
-        return video, audio, label
+        if self.transform is not None:
+            video = self.transform(video)
+
+        # return video, audio, label
+        return video, label
```

Comment:
If you are gathering the results, wouldn't you need an index for validation as well?

Reply:
Yes, but I'm currently making an approximation during evaluation, where I can have some images / clips which are accumulated multiple times.

I'd rather not return the `index` in the dataset itself, as this is a kind of a hack that I think shouldn't live in torchvision.

If we definitely want to have a more accurate way of handling distributed evaluation, then I think I'll add an extra dataset in `references` that just wraps the `idx` as well, doing something like

```python
class Kinetics2(Kinetics):
    def __getitem__(self, idx):
        video, label = super(Kinetics2, self).__getitem__(idx)
        return video, label, idx
```

Reply:
Improving the validation to not take into account duplicate indices will be done in a follow-up PR.